Source/WebKit/UIProcess/Extensions/WebExtension.cpp

```diff
@@ -524,6 +530,62 @@ Vector<Ref<API::Error>> WebExtension::errors()
     return m_errors;
 }
 
+const Vector<String>& WebExtension::supportedLocales()
+{
+    if (!m_supportedLocales.isEmpty())
+        return m_supportedLocales;
+
+    auto localesString = "_locales/"_s;
+    auto localeDirectoryPath = resourceFileURLForPath(localesString).fileSystemPath();
+    if (!localeDirectoryPath.isEmpty()) {
+        m_supportedLocales = FileSystem::listDirectory(localeDirectoryPath);
+        return m_supportedLocales;
+    }
+
+    // For tests that don't have a file system location, check the resource cache.
```

Comment:
Do we want to check to make sure we are in some kind of testing mode here and return early if not?

Reply:
Nah, we use `m_resources` for non-testing caches too.